### ISSUE TITLE
Accessing LTR/STTB/CMR

### ISSUE THREAD

**Issue by joe-pitt** [ICOS-Carbon-Portal/pylib#195]
Lots of important information is stored in the variables LTR, STTB and CMR. This data is retrieved by dobj for the "Obspack" .nc files, e.g.:
```
from icoscp.dobj import Dobj
dobj = Dobj("https://meta.icos-cp.eu/objects/ILHK0NhgL5-yyk57BxgSzuxf")
dobj.get()
```

However, for most other files this info isn't retrieved, e.g.:
```
dobj = Dobj("https://meta.icos-cp.eu/objects/tpXGpHAV7VHxrypNTmp__rMN")
dobj.get()
```
If you click on the url in the second example and download the ASCII file, the LTR, STTB and CMR data is there. Could the option to retrieve this data be added?

**Comment by joe-pitt** [comment]
On a similar note, for the new FastTrack files (e.g. https://meta.icos-cp.eu/objects/hI04Z_fR5-XvEr5ib-3c_RT9) it seems like Stdev is missing too.

**Comment by ukarst** [comment]
Thanks for your feedback. Indeed, it seems that for some data types the Stdev is not part of the "previewable vaiables" and therefore not available using our icoscp python library. We will follow up on this right after the Christmas break.

**Comment by joe-pitt** [comment]
Thanks for looking into this! I can see that this wouldn't be worth the effort for the L2 and NRT datasets - I will use the Obspack collection in that case. But is there any chance the specification for the FastTrack and EYE-AVE-PAR files could be changed? Without the stdev and uncertainty these data really can't be used in any inversions, and the only way to get at these currently would appear to be manually downloading the files and using some separate code to read them in.

**Comment by ukarst** [comment]
Good point! 
As this would require an executive decision, I will bring this up with my colleagues.

**Comment by joe-pitt** [comment]
That would be great thanks!

**Comment by atverm** [comment]
You can use the most recent netcdf Obspack time series that is a composition of historic, ICOS L2 ,  Fasttrack L1.5 and NRT L1 data, it includes the three uncertainty columns for  SMR, STTB, LTR as available for L2 and L1.5. And Stdev but that I would not use for data selection. The data is listed here:

[(https://data.icos-cp.eu/portal/#{"filterCategories":{"theme":["atmosphere"],"format":["netcdfTimeSeries"]}})]

**Comment by joe-pitt** [comment]
Oh yeah I didn't spot that the Fasttrack data was in there. In that case I guess I can live with the situation as it currently is (I will just download the EYE-AVE-PAR files and read them in), but for future datasets I think it would be nice to include as many columns in the spec as possible.

@atverm - am I right in thinking that there will soon be some new netcdf Obspack time series files on the CP running up to January 2025 (with Fasttrack data up to September 2024)? I can see the ASCII files for this Fasttrack release went up on Monday, but I can't see the netcdf files yet

**Comment by atverm** [comment]
Yes, we will update the Fasttrack product somewhere next week maybe. We are a bit pressed with time to do the processing at the moment.

Alex

**Comment by brendan-m-murphy** [comment]
Just a note to anyone else with this issue: you can get the full text file using `icoscp_core` using:

```python
import io
import re
from zipfille import ZipFile

import pandas as pd
from icoscp_core import data

# save to disk
data.save_to_folder(dobj_uri, folder_path)

# or download into memory
# this method only works if if it a zipped text file
_, resp = data.get_file_stream(dobj_uri)

with ZipFile(io.BytesIO(resp.read()), "r") as z:
    with z.open(z.infolist()[0].filename) as f:
        icos_text = f.read().decode("utf-8")

# read with pandas, assuming ascii ATC time series format
# first need to get length of header
pat = re.compile(r"^# HEADER LINES: (\d+)")
m = pat.search(icos_text)
if m:
    header_lines = int(m.group(1))
else:
    header_lines = len([line for line in icos_text.split("\n") if line.startswith("#")])

df = pd.read_csv(io.StringIO(icos_text), skiprows=header_lines-1, sep=";")
df.columns = [col.removeprefix("#") for col in df.columns]  # first col has '#' from comments

```

You can also download the netCDF time series in a similar way, but you don't need to unzip the file, and you can read the response like `xr.open_dataset(io.BytesIO(resp.read()))`.

As an aside: the obspack label data variable is quite big, but it looks like a fixed string followed by integers. Can ICOS store the fixed string separately and just have a data variable with the varying integers?
